docs/CIP_5_Large_Batch_Handling_Improvements.md

```diff
@@ -0,0 +1,56 @@
+# CIP-5: Large Batch Handling Improvements Proposal
+
+## Status
+
+Current Status: `Under Discussion`
+
+## **Motivation**
+
+As users start putting Chroma in its paces and storing ever-increasing datasets, we must ensure that errors
+related to significant and potentially expensive batches are handled gracefully. This CIP proposes to add a new
+setting, `max_batch_size` API, on the local segment API and use it to split large batches into smaller ones.
+
+## **Public Interfaces**
+
+The following interfaces are impacted:
+
+- New Server API endpoint - `/pre-flight-checks`
+- New `max_batch_size` property on the `API` interface
```

Comment:
Naming thought: Is it confusing that this only applies to add?

Reply:
Upsert too, gonna add that. How about get/delete - maybe future PR?

Reply:
added it to all three - add, update, upsert